presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -65,68 +61,77 @@
 
+            KafkaConsumer<ByteBuffer, ByteBuffer> consumer = consumerManager.createConsumer(Thread.currentThread().getName(), node);
+            List<PartitionInfo> partitions = consumer.partitionsFor(topic);
             ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
 
-            for (TopicMetadata metadata : topicMetadataResponse.topicsMetadata()) {
-                for (PartitionMetadata part : metadata.partitionsMetadata()) {
-                    log.debug("Adding Partition %s/%s", metadata.topic(), part.partitionId());
-
-                    Broker leader = part.leader();
-                    if (leader == null) {
-                        throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Leader election in progress for Kafka topic '%s' partition %s", metadata.topic(), part.partitionId()));
-                    }
-
-                    HostAddress partitionLeader = HostAddress.fromParts(leader.host(), leader.port());
-
-                    SimpleConsumer leaderConsumer = consumerManager.getConsumer(partitionLeader);
-                    // Kafka contains a reverse list of "end - start" pairs for the splits
-
-                    long[] offsets = findAllOffsets(leaderConsumer, metadata.topic(), part.partitionId());
-
-                    for (int i = offsets.length - 1; i > 0; i--) {
-                        KafkaSplit split = new KafkaSplit(
-                                connectorId,
-                                metadata.topic(),
-                                kafkaTableHandle.getKeyDataFormat(),
-                                kafkaTableHandle.getMessageDataFormat(),
-                                kafkaTableHandle.getKeyDataSchemaLocation().map(KafkaSplitManager::readSchema),
-                                kafkaTableHandle.getMessageDataSchemaLocation().map(KafkaSplitManager::readSchema),
-                                part.partitionId(),
-                                offsets[i],
-                                offsets[i - 1],
-                                partitionLeader);
-                        splits.add(split);
-                    }
+            for (PartitionInfo partition : partitions) {
+                Node leader = partition.leader();
+                if (leader == null) {
+                    throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Leader election in progress for Kafka topic '%s' partition %s", topic, partition.partition()));
+                }
+
+                HostAddress partitionLeader = HostAddress.fromParts(leader.host(), leader.port());
+                long startTimestamp = layoutHandle.getStartOffsetTimestamp();
+                long endTimestamp = layoutHandle.getEndOffsetTimestamp();
+
+                if (startTimestamp > endTimestamp) {
+                    throw new IllegalArgumentException(String.format("Invalid Kafka Offset start/end pair: %s - %s", startTimestamp, endTimestamp));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: endTimestamp (declared at line 107), startTimestamp (declared at line 106). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 110. Consider formatting this data with &#39;%d&#39; instead.